### DISCUSSION TITLE
Typescript: How to change `parent` typing for chained resolver functions?

### DISCUSSION THREAD

**Discussion by mryanlo** [dotansimha/graphql-code-generator#6675] [General] [answered]
I am trying to make a gql resolver query so that we only query for certain fields if the user asked for it.

For example:
```
// In .graphql file
type User {
  id: String!
  name: String!
  address: Address // here, getting Address is expensive so we have to separate the query into 2 API calls
}

type Address {
  street
  city
  zip
}



// in resolver.ts file:
interface UserExtension extends User {
  raw: BackendType.UserRaw
}
const Resolvers = {
 Query: {
   getUser(_, id, _) { 
      const result = ...; // call backend API to get user details without Address field. result is not same type as User in gql file
      return { ..., raw: result }; // pass in raw backend result so the email resolver can reference it
   }
 }
 User: {
     email(parent: UserExtension ...) { // with strict function type check this fails
        /// return new object of type User
     }
  }
}
```

but as you see in the comment, the strict function type check fails because we're expecting `parent` parameter for the email field to be of type User whereas I want it to be UserExtension so we can extract the original data from the raw backend result. Is there a way to force typescript to accept this without the `any` typing?

**Comment by ardatan** [comment, marked as answer]
You can use `mappers` in the config;
https://www.graphql-code-generator.com/docs/plugins/typescript-resolvers

Also this would be helpful;
https://the-guild.dev/blog/better-type-safety-for-resolvers-with-graphql-codegen
